### PR TITLE
[READY] tgui

### PR DESCRIPTION
**Pull request by neersighted**
Announcing... tgui!

NanoUI has now been rewritten, frontend and backend -- thus, it now has a new name.

This is hopefully the last (heh!) ever rewrite. At this point most of the work to be done is documentation and converting stringbuilder UIs.

It kind of changes a lot -- UIs are the same as before, differences are mostly cosmetic.
The immediate difference is more consistent styles and animated UIs (numbers and bars and buttons).
The animations are purely visual and I will probably add the option to disable them (the fancy pref) if people complain. They work great even on IE8 however.

This update should also fix compatibility with IE8; as I am now using a framework to keep things sane.

Mostly, this rewrite opens up many possibilities for future UIs, such as minimaps and graphs, as each UI can have code and styles specific to it; as well as breaking up large UIs into multiple files (Air Alarms are like this).

So with that, all hail the era of tgui!

:cl:
tweak: NanoUI is now tgui (aka NanoUI 4)
experiment: Rewrote NanoUI again... everything should be fixed; hopefully.
:cl:

### CODE REVIEW

**Comment by GunHog** [comment]
At least provide a log of what you did.

**Comment by optimumtact** [comment]
read the code

File: nano/gulp/tasks/js.coffee [hunk 1]
```diff
@@ -0,0 +1,41 @@
+c = require "../config"
+f = c.flags
+p = c.paths
+b = c.plugins.browserify
+g = c.plugins.gulp
+
+
+browserify = require "browserify"
+gulp       = require "gulp"
```
Comment (KazeEspada) [inline nano/gulp/tasks/js.coffee]: GULP!

**Comment by Shadowlight213** [comment]
@optimumtact 
>normal coders being able to read nanoui code

We aren't that good.

**Comment by neersighted** [comment]
Just gonna put this here...

**Comment by neersighted** [comment]
Also for comparison with OG nanoui:

**Comment by Shadowlight213** [comment]
Oh my god that new updating is amazing. :+1:

**Comment by neersighted** [comment]
It happening!!!!

**Pull request merged**